lib/intercept.js

```diff
@@ -235,6 +235,27 @@ function ErroringClientRequest(error) {
 
 inherits(ErroringClientRequest, http.ClientRequest)
 
+function urlToOptions(input) {
```

Comment:
Is there some API and/or source code reference for this translation?

Reply:
Yes there is, it's from the Node.js' `internal/url.js` file. https://github.com/nodejs/node/blob/master/lib/internal/url.js#L1257

Added some comment to the `urlToOptions` function, and the link.